features/nanostack/nanostack-interface/Nanostack.cpp

```diff
@@ -150,6 +158,50 @@ static int8_t find_interface_by_address(const uint8_t target_addr[16])
     return -1;
 }
 
+static int8_t nanostack_interface_id_parse(const char* interface_name)
+{
+    int8_t interface_id = -1;
+
+    TRACE_DEEP("nanostack_interface_id_parse() %s", interface_name ? interface_name : "null");
+
+    if (!interface_name) {
+        return -1;
+    }
+
+    // parse interface ID from the interface_name
+    if (strlen(interface_name) < 4) {
+        return -1;
+    }
+
+    interface_id = atoi(&interface_name[3]);
```

Comment:
Do all interfaces have exactly 4 leters ? Mesh ETH? TUN? is there possibility to search first number and then atoi it instead of assuming

Reply:
I agree, this is problematic. Interface name is created https://github.com/ARMmbed/mbed-os/blob/feature-wisun/features/nanostack/mbed-mesh-api/source/MeshInterfaceNanostack.cpp#L132 and but it can also be given from application.

Reply:
Application returns interface name that the nanostack interfaces have returned, so nanostack is in full control of those. Nanostack interface naming pattern is to have always 3 letters and a number.